mapping.json

```diff
@@ -1,38 +1,48 @@
-        ["google-analytics.com/gtm/js", "gtm.js"]
+        { "url": "google-analytics.com/ga.js", "surrogate": "ga.js" },
+        { "url": "google-analytics.com/analytics.js", "surrogate": "analytics.js" },
+        { "url": "google-analytics.com/inpage_linkid.js", "surrogate": "inpage_linkid.js" },
+        { "url": "google-analytics.com/plugins/ga/inpage_linkid.js", "surrogate": "inpage_linkid.js" },
+        { "url": "google-analytics.com/cx/api.js", "surrogate": "api.js" },
+        { "url": "google-analytics.com/gtm/js", "surrogate": "gtm.js" }
     ],
     "googletagmanager.com": [
-        ["googletagmanager.com/gtm.js", "gtm.js"]
+        { "url": "googletagmanager.com/gtm.js", "surrogate": "gtm.js" }
     ],
     "googletagservices.com": [
-        ["googletagservices.com/gpt.js", "gpt.js"],
-        ["googletagservices.com/tag/js/gpt.js", "gpt.js"]
-    ],        
+        { "url": "googletagservices.com/gpt.js", "surrogate": "gpt.js" },
+        { "url": "googletagservices.com/tag/js/gpt.js", "surrogate": "gpt.js" }
+    ],
     "googlesyndication.com": [
-        ["googlesyndication.com/adsbygoogle.js", "adsbygoogle.js"],
-        ["googlesyndication.com/pagead/js/adsbygoogle.js", "adsbygoogle.js"]
+        { "url": "googlesyndication.com/adsbygoogle.js", "surrogate": "adsbygoogle.js" },
+        { "url": "googlesyndication.com/pagead/js/adsbygoogle.js", "surrogate": "adsbygoogle.js" }
     ],
     "doubleclick.net": [
-        ["doubleclick.net/instream/ad_status.js", "ad_status.js"],
-        ["doubleclick.net/tag/js/gpt.js", "gpt.js"]
+        { "url": "doubleclick.net/instream/ad_status.js", "surrogate": "ad_status.js" },
+        { "url": "doubleclick.net/tag/js/gpt.js", "surrogate": "gpt.js" }
     ],
     "scorecardresearch.com": [
-        ["scorecardresearch.com/beacon.js", "beacon.js"]
+        { "url": "scorecardresearch.com/beacon.js", "surrogate": "beacon.js" }
     ],
     "outbrain.com": [
-        ["outbrain.com/outbrain.js", "outbrain.js"]
+        { "url": "outbrain.com/outbrain.js", "surrogate": "outbrain.js" }
     ],
     "amazon-adsystem.com": [
-        ["amazon-adsystem.com/aax2/amzn_ads.js", "amzn_ads.js"]
+        { "url": "amazon-adsystem.com/aax2/amzn_ads.js", "surrogate": "amzn_ads.js" }
     ],
     "chartbeat.com": [
-        ["chartbeat.com/chartbeat.js", "chartbeat.js"],
-        ["chartbeat.com/js/chartbeat.js", "chartbeat.js"]
+        { "url": "chartbeat.com/chartbeat.js", "surrogate": "chartbeat.js" },
+        { "url": "chartbeat.com/js/chartbeat.js", "surrogate": "chartbeat.js" }
+    ],
+    "facebook.net": [
+        { "regexUrl": "facebook\\.net\\/[a-z_A-Z]+\\/sdk\\.js", "surrogate": "fb-sdk.js", "action": "block-ctl-fb" },
+        { "regexUrl": "facebook\\.net\\/[a-z_A-Z]+\\/all\\.js", "surrogate": "fb-sdk.js", "action": "block-ctl-fb" }
+    ],
+    "facebook.com": [
+        { "regexUrl": "facebook\\.com\\/[a-z_A-Z]+\\/sdk\\.js", "surrogate": "fb-sdk.js", "action": "block-ctl-fb" }
```

Comment:
again, missing some details from TD discussions - do we need a separate action for each CTL target? This means that whenever we add a new target (instagram, twitter, etc.) we will have to update client apps to understand this new action.

Reply:
This was discussed in https://app.asana.com/0/0/1201397262498310/1201415252824341/f https://app.asana.com/0/0/1201397262498310/1201415252824341/f

Basically these are per-client implementations anyway (each needs bespoke support whether it be new content scripts, CBR rules, etc).

The intent is to update these clients to ignore rules with unrecognized actions, and then trigger the corresponding code path for CTL once we port over the current implementations to rely on the TDS rules instead of their own.

Reply:
+1 to a new action for those, but I wonder why those actions have to be youtube/facebook specific? Ideally all configuration related to new CTL targets would be separate from client implementations allowing us to rapidly add more targets. You mention:

- content scripts - but those are the surrogates, right? Those can be updated independently from the browser (remotely / as a dependency)
- CBR rules - those should be generated based on blocklist which is also remote / dependency

Asking differently - if we add Instagram CTL support tomorrow, what will have to happen on clients?

Reply:
Yes because each CTL features has needed some bespoke client support (native code) to support it.  We could also use a generic "clicktoload" action that covers features that don't need extra work, right?  TBH I feel like we're rehashing the TD here.

Reply:
"needed some bespoke client support" - I'm curious what is it, but you are totally right, this is not a PR but a TD discussion so I'm bringing it back there.